helix-core/src/main/java/org/apache/helix/controller/stages/MessageGenerationPhase.java

```diff
@@ -344,6 +367,13 @@ private void schedulePendingMessageCleanUp(
             if (accessor.removeProperty(msg.getKey(accessor.keyBuilder(), instanceName))) {
               LogUtil.logInfo(logger, _eventId, String
                   .format("Deleted message %s from instance %s", msg.getMsgId(), instanceName));
+              if (staleMessageMap != null && staleMessageMap.containsKey(msg.getTgtName())
```

Comment:
I think staleMessageMap won't be null.

Reply:
I think it won't be. It's got instantiated in `CurrentStateOutput`